<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>940</width>
    <height>634</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="0">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="manual_tab">
       <attribute name="title">
        <string>Manual Control</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <widget class="QLabel" name="image_robotic_arm">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="pixmap">
           <pixmap resource="resource.qrc">:/images/robot_arm.jpg</pixmap>
          </property>
          <property name="scaledContents">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_9">
            <item>
             <widget class="QCheckBox" name="checkbox_joystick">
              <property name="text">
               <string>Joystick</string>
              </property>
              <property name="tristate">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_joystick_status">
              <property name="text">
               <string>Joystick: Not Activate</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QCheckBox" name="checkbox_gripper">
              <property name="text">
               <string>Gripper</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="button_home">
              <property name="text">
               <string>Home</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QLabel" name="label_base">
              <property name="text">
               <string>Base</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QPushButton" name="button_base_minus">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="resource.qrc">
                  <normaloff>:/images/minus_.png</normaloff>:/images/minus_.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBox_base">
                <property name="keyboardTracking">
                 <bool>false</bool>
                </property>
                <property name="maximum">
                 <number>180</number>
                </property>
                <property name="value">
                 <number>90</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="button_base_plus">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="resource.qrc">
                  <normaloff>:/images/plus.png</normaloff>:/images/plus.png</iconset>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QLabel" name="label_shoulder">
              <property name="text">
               <string>Shoulder</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QPushButton" name="button_shoulder_minus">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="resource.qrc">
                  <normaloff>:/images/minus_.png</normaloff>:/images/minus_.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBox_shoulder">
                <property name="keyboardTracking">
                 <bool>false</bool>
                </property>
                <property name="maximum">
                 <number>180</number>
                </property>
                <property name="value">
                 <number>180</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="button_shoulder_plus">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="resource.qrc">
                  <normaloff>:/images/plus.png</normaloff>:/images/plus.png</iconset>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QLabel" name="label_elbow">
              <property name="text">
               <string>Elbow</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QPushButton" name="button_elbow_minus">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="resource.qrc">
                  <normaloff>:/images/minus_.png</normaloff>:/images/minus_.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBox_elbow">
                <property name="keyboardTracking">
                 <bool>false</bool>
                </property>
                <property name="maximum">
                 <number>180</number>
                </property>
                <property name="value">
                 <number>5</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="button_elbow_plus">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="resource.qrc">
                  <normaloff>:/images/plus.png</normaloff>:/images/plus.png</iconset>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QLabel" name="label_wrist">
              <property name="text">
               <string>Wrist</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QPushButton" name="button_wrist_minus">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="resource.qrc">
                  <normaloff>:/images/minus_.png</normaloff>:/images/minus_.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBox_wrist">
                <property name="keyboardTracking">
                 <bool>false</bool>
                </property>
                <property name="maximum">
                 <number>180</number>
                </property>
                <property name="value">
                 <number>90</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="button_wrist_plus">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="resource.qrc">
                  <normaloff>:/images/plus.png</normaloff>:/images/plus.png</iconset>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <widget class="QLabel" name="label_gripper">
              <property name="text">
               <string>Gripper</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QPushButton" name="button_gripper_minus">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="resource.qrc">
                  <normaloff>:/images/minus_.png</normaloff>:/images/minus_.png</iconset>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spinBox_gripper">
                <property name="keyboardTracking">
                 <bool>false</bool>
                </property>
                <property name="maximum">
                 <number>180</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="button_gripper_plus">
                <property name="text">
                 <string/>
                </property>
                <property name="icon">
                 <iconset resource="resource.qrc">
                  <normaloff>:/images/plus.png</normaloff>:/images/plus.png</iconset>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="head_tab">
       <attribute name="title">
        <string>Head Control</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_14">
        <item>
         <widget class="QLabel" name="label_camera">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>640</width>
            <height>480</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: black; color: white; border: 2px solid red;</string>
          </property>
          <property name="text">
           <string>Camera OFF</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="voice_tab">
       <attribute name="title">
        <string>Voice Control</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <widget class="QLabel" name="label_voice_status">
          <property name="text">
           <string>TextLabel</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QPlainTextEdit" name="text_voice_log">
            <property name="plainText">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_8">
            <item>
             <widget class="QPlainTextEdit" name="text_voice_input">
              <property name="plainText">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="text_command_sheet">
              <property name="plainText">
               <string>Voice Command Guide

Commands:
• “Pick / Take / Get” &gt; Pick up food
• “Feed / Send” &gt; Bring food to your mouth
• “Home” &gt; Move robotic arm to home position
• “Finish” &gt; Finish eating
• “Head” &gt; Switch to head-control mode

Confirmation:
• &quot;Yes&quot; &gt; yes, yeah, yep, yup, sure
• &quot;No&quot;&gt; no, nope, nah
</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>940</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections/>
</ui>
